cnf ran: Add siteconfig operator pod restart and cluster instance delete tests

mpmaruthu opened the pull request:
@mcornea , @klaskosk , Please review this PR when you have a time. 
cc: @yliu127 , @dgonyier 

Thanks.
/Peri M

Adding siteconfig-operator test case - [OCP-75374](https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-75374)

Adding siteconfig-operator test case - [OCP-75376](https://polarion.engineering.redhat.com/polarion/redirect/project/OSE/workitem?id=OCP-75376)

Adding siteconfig-operator test case [OCP-75411](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-75411)

Sample test configuration for your reference:
http://registry.kni-qe-0.lab.eng.rdu2.redhat.com:3000/kni-qe/ztp-site-configs/src/helix54+helix77-4.17/siteconfig/ztp-test/siteconfig-operator/detach-ai-sno/kustomization.yaml#L4


Manual test logs:- 
Test case 1: https://docs.google.com/document/d/1ihBMjxsZmi2RRX90f5a--vmqzb0hDUxWQIlK4t8Wugk/edit?usp=sharing

Test case 2: https://docs.google.com/document/d/1FsTuWvS0UGaCujsfy5UQRJIlU69T2TdbGJ6Jt6OFANA/edit?usp=sharing

Test case 3: https://docs.google.com/document/d/1luC2QAEUHslWQ136ZvZXwODm38oLV-gFAmwOOwLTMP4/edit?usp=sharing

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,164 @@
+		clusterDeployment, err := hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull cluster deployment manifests")
+		if err == nil {
+			err := clusterDeployment.Exists()
```
also `builder.Exists()` returns a bool so the expect will not work as expected

mpmaruthu replied:
Okay, updated. Thanks!

klaskosk commented on tests/cnf/ran/gitopsztp/internal/tsparams/consts.go:
```diff
@@ -83,6 +85,8 @@ const (
 	// ZtpKustomizationPath is the path to the kustomization file in the ztp test.
 	ZtpKustomizationPath = "/kustomization.yaml"
+	// ZtpTestPathDetachAISNO is the git path for the siteconfig operator detach AI SNO cluster instance test.
+	ZtpTestPathDetachAISNO = "ztp-siteconfigoperator-test/detach-ai-sno/"
```
for consistency, the path should be `ztp-test/siteconfig-operator/detach-ai-sno` that way we have one directory for all ztp tests
also nit: this variable should be declared above ZtpKustomizationPath

mpmaruthu replied:
Okay, Noted for future use. I will remove this test directory structure definition here.
The reason for removal will be followed,

1. The test steps what I have developed here are common to detach-ai-sno and destach-ai-mno
2. Hence, i would like to approach more generic methodology here.
3. I will update the respective CI test profile to have and reflect the  test configuration directory structure required to verify detaching site resources at the root level.

Thanks,

mpmaruthu replied:
Done.

mpmaruthu commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,164 @@
+
+	// 75376 - Detaching the AI single-node openshift (SNO) spoke cluster.
+	It("Validate detaching the AI single-node openshift (SNO) spoke cluster", reportxml.ID("75376"), func() {
+		// Update the ztp-siteconfigoperator-test git path.
+		gitDetails := tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName]
+		testGitPath := gitdetails.JoinGitPaths([]string{
+			gitDetails.Path,
+			tsparams.ZtpTestPathDetachAISNO,
+		})
+
+		By("checking if the git path exists")
+		if !gitdetails.DoesGitPathExist(gitDetails.Repo, gitDetails.Branch, testGitPath+tsparams.ZtpKustomizationPath) {
+			Skip(fmt.Sprintf("git path '%s' could not be found", testGitPath))
+		}
+
+		gitDetails.Path = testGitPath
+
+		// Detaching site resources in root level kustimozation.yaml.
+		By("updating the Argo CD clusters app with the detach AI SNO cluster instance git path")
+		err := gitdetails.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdClustersAppName,
+			gitDetails,
+			true,
+			true)
+		Expect(err).ToNot(HaveOccurred(), "Failed to update the Argo CD clusters app with new git details")
```
Okay, Noted for the future use. Thanks!
I will remove the pattern written here.
The reason for removal will be followed,

- The test steps what I have developed here are common to [detach-ai-sno](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-75376) and [destach-ai-mno](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-75374).

- Hence, i would like to approach more generic methodology here.
- I will update the respective CI test profile to have and reflect the test configuration directory structure required to verify detaching site resources at the root level.

Sample reference here to know more,
https://gitlab.cee.redhat.com/ocp-edge-qe/ocp-edge-ci/-/blob/master/jenkins/ci-profiles-new/4.17/far-edge-vran-ran-ibu.yaml?ref_type=heads#L136-201

mpmaruthu replied:
Done.

mpmaruthu commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,164 @@
+
+		// Test expected results validation.
+		By("checking the infra env manifests removed on hub")
+		infraEnv, err := assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to pull infra env manifests")
+		if err == nil {
+			err := infraEnv.Exists()
+			Expect(err).To(HaveOccurred(), "Found spoke infra env manifests but expected to be removed")
+		}
```
Okay, updated. Thanks!

mpmaruthu commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,164 @@
+		}
+
+		By("checking the bare metal host manifests removed on hub")
+		bareMetalHost, err := bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
```
Okay, updated. Thanks!

mpmaruthu commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,164 @@
+		}
+
+		By("checking the cluster deployment manifests removed on hub")
+		clusterDeployment, err := hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
```
Okay, updated. Thanks!

klaskosk commented on tests/cnf/ran/internal/ranparam/const.go:
```diff
@@ -35,6 +35,10 @@ const (
 	RetryInterval = 10 * time.Second
 	// RetryCount retry count for node exec commands.
 	RetryCount = 3
+	// SNO represents spoke cluster type as single-node openshift (SNO) cluster.
+	SNO = "sno"
+	// Standard represents spoke cluster type as multi-node openshift (MNO) cluster.
+	Standard = "standard"
```
Could you make the cluster type into a separate type like what HubOperatorName does?

mpmaruthu replied:
Okay, updated. Thanks!

klaskosk commented on tests/cnf/ran/internal/rancluster/rancluster.go:
```diff
@@ -125,3 +126,17 @@ func WaitForNumberOfNodes(client *clients.Settings, expected int, timeout time.D
+		return ranparam.SNO, nil
+	}
+
+	return ranparam.Standard, fmt.Errorf("could not determine spoke cluster type")
```
for the error case, could you make it return `""` instead of standard?

mpmaruthu replied:
Okay, updated. Thanks!

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-operator.go:
```diff
@@ -0,0 +1,219 @@
+		// Test step 1-Recreate default assisted installer template reference ConfigMap CR. I.e. the one recently deleted.
+
+		// Test step 2-Recreate the ClusterInstance custom resource. I.e. the one recently deleted.
+		By("resetting the clusters app back to the original settings")
+		err = gitdetails.SetGitDetailsInArgoCd(
+			tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName], true, false)
+		Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+
+		// Test step 2 expected results validation.
+		By("checking the infra env manifests exists on hub")
+		_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke infra env manifests")
+
+		By("checking the bare metal host manifests exists on hub")
+		_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke bmh manifests")
+
+		By("checking the cluster deployment manifests exists on hub")
+		_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke cluster deployment manifests")
+
+		By("checking the NM state config manifests exists on hub")
+		nmStateConfigList, err = assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+		Expect(nmStateConfigList).ToNot(BeEmpty(), "Failed to find NM state config manifests")
+
+		By("checking the klusterlet addon config manifests exists on hub")
+		_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke kac manifests")
+
+		By("checking the agent cluster install manifests exists on hub")
+		_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+		Expect(err).ToNot(HaveOccurred(), "Failed to find spoke agent cluster install manifests")
```
are these the same between the test cases? maybe they should go into an AfterEach

mpmaruthu replied:
Okay, updated. Thanks!

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,182 @@
+			Expect(err).ToNot(HaveOccurred(), "Failed to update Argo CD clusters app with new git path")
+
+			// Test step 2 expected results validation.
+			By("checking the infra env manifests removed on hub")
+			_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke infra env manifests but expected to be removed")
+
+			By("checking the bare metal host manifests removed on hub")
+			_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke bmh manifests but expected to be removed")
+
+			By("checking the cluster deployment manifests removed on hub")
+			_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke cluster deployment manifests but expected to be removed")
+
+			By("checking the NM state config manifests removed on hub")
+			nmStateConfigList, err := assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+			Expect(nmStateConfigList).To(BeEmpty(), "Found spoke NM state config manifests but expected to be removed")
+
+			By("checking the klusterlet addon config manifests removed on hub")
+			_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke kac manifests but expected to be removed")
+
+			By("checking the agent cluster install manifests removed on hub")
+			_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke ACI manifests but expected to be removed")
+
+			By("verifying installed spoke cluster should still be functional")
+			_, err = version.GetOCPVersion(Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get OCP version from spoke and verify spoke cluster access")
```
if these are the same between the two cases, they could be moved to a function at the end of this file

mpmaruthu replied:
Okay,updated. Thanks!

klaskosk commented on tests/cnf/ran/internal/ranhelper/ranhelper.go:
```diff
@@ -59,3 +62,13 @@ func ExecLocalCommand(timeout time.Duration, command string, args ...string) (st
+	glog.V(ranparam.LogLevel).Infof("podList[0] matching podLabelSelector is '%v'",
+		podList[0].Definition.Name)
+
+	return podList[0].Definition.Name, err
```
This will panic if no pods exist with the label or the list function returns an error

mpmaruthu replied:
Good catch. Thanks!
I have updated the function and checked it by running dry-run validation. It's passed on both scenarios - [1] pod existence and [2] pod non-existence cases.

**pod non-existence case validation logs:**
>   Validate detaching the AI multi-node openshift spoke cluster [75374, test_id:75374]
>   /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77
>   > Enter [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer cluster level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:90 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer node level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:99 @ 09/25/24 17:37:17.832
>   END STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:37:17.968 (255ms)
>   STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:37:17.969
>   STEP: Get the siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:131 @ 09/25/24 17:37:17.969
>   END STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:37:18.093 (125ms)
>   [FAILED] Failed to get siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller from rhacm namespace
>   Unexpected error:
>       <*errors.errorString | 0xc000d56270>: 
>       no pod found with label app.kubernetes.io/name=siteconfig-controller under namespace rhacm
>       {
>           s: "no pod found with label app.kubernetes.io/name=siteconfig-controller under namespace rhacm",
>       }
>   occurred
>   In [It] at: /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:134 @ 09/25/24 17:37:18.093
>   < Exit [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:37:18.093 (380ms)


**pod existence case validation logs:**
>  Validate detaching the AI multi-node openshift spoke cluster [75374, test_id:75374]
>   /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77
>   > Enter [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer cluster level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:90 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer node level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:99 @ 09/25/24 17:39:40.67
>   END STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:39:41.206 (920ms)
>   STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:39:41.206
>   STEP: Get the siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:131 @ 09/25/24 17:39:41.206
> pod name%!(EXTRA string=siteconfig-controller-manager-665757d8d8-kfg7w)  STEP: Pull the siteconfig operator pod name from namespace rhacm - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:138 @ 09/25/24 17:39:41.413
>   END STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:39:44.074 (2.868s)
> 

new patch reflecting the updated changes.

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,229 @@
+
+// ValidateAISpokeClusterInstallCRsRemoved verifies AI spoke cluster install CRs removed and spoke cluster accessible.
+func ValidateAISpokeClusterInstallCRsRemoved() {
+	By("checking the infra env manifests removed on hub", func() {
```
for consistency, could you please switch this PR to use the form of `By` without the `func() {}`?

mpmaruthu replied:
Okay, updated. Thanks!

klaskosk replied:
@mpmaruthu I don't see the update here, it looks like there's still a lot that use `func() {}`

mpmaruthu replied:
Yes, I have updated in my local feature branch and validating the changes by running dry-run. Will keep you posted once the new patch is available for review. Thanks!

klaskosk replied:
@mpmaruthu Ah that makes sense, I'll be on the lookout

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,229 @@
+				By("deleting default assisted installer cluster level templates ConfigMap CR")
+				clusterTemplateConfigMap, err := configmap.Pull(HubAPIClient, tsparams.DefaultAIClusterTemplatesConfigMapName,
+					ranparam.AcmOperatorNamespace)
+				Expect(err).ToNot(HaveOccurred(), "Failed to find default AI cluster level templates config map")
+
+				if err == nil {
```
since you assert the error didn't occur, it is always `nil` so the if statement is unnecessary. Is it a test failure if the cm does not exist?

mpmaruthu replied:
Is it a test failure if the cm does not exist? **re: `no, i am deleting default cluster level templates ConfigMap custom resource after spoke cluster installed, and then make sure the spoke cluster is still accessible`**

mpmaruthu replied:
since you assert the error didn't occur, it is always nil so the if statement is unnecessary. re: `I am deleting ConfigMap custom resource after asserting err == nil i.e. first I am trying to find default cluster level templates ConfigMap customer resource exists in rhacm namespace, and then delete it if it exists`

klaskosk replied:
@mpmaruthu In that case then, I think you want to remove the `Expect` line since that will fail the test if the resource does not exist. This goes for all of the places where you're trying to delete if it does not exist

mpmaruthu replied:
updated. Thanks!

klaskosk commented on .golangci.yml:
```diff
@@ -95,6 +95,7 @@ linters-settings:
 linters:
   disable:
     - ireturn
+    - wsl
```
please remove the changes to the linters

mpmaruthu replied:
done.

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,216 @@
+			}
+
+			// Test step 1-Delete default assisted installer template reference ConfigMap CRs after spoke cluster installed.
+			By("deleting default assisted installer template reference ConfigMap custom resources", func() {
```
could you remove the `func() {}` inside this test case too please?

mpmaruthu replied:
done.

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,216 @@
+
+			// Test teardown.
+			// Recreate default assisted installer template reference ConfigMap CRs by deleting siteconfig operator pod.
+			By("deleting siteconfig operator pod running under rhacm namespace on hub cluster. ", func() {
```
see above about the By

mpmaruthu replied:
done.

klaskosk commented on tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:
```diff
@@ -0,0 +1,216 @@
+
+			// Teardown test expected results validation.
+			// The default assisted installer template reference ConfigMap custom resource should be recreated successfully.
+			By("checking the default assisted installer template reference ConfigMap CRs recreated successfully.", func() {
```
see above about the By

mpmaruthu replied:
done.

klaskosk reviewed (commented):
A couple minor comments otherwise lgtm

klaskosk commented on .golangci.yml:
```diff
@@ -141,7 +141,6 @@ linters:
     - unparam
     - unused
     - varnamelen
-    - wsl
```
please remove this change to the lint config too

mpmaruthu replied:
Updated, Thanks!

klaskosk reviewed (approved):
/lgtm

mpmaruthu commented:
@klaskosk . @josclark42 , I have updated this PR with minor changes and added additional test validation check points to improve and make the test automation code more robust. Thanks!

The reason for changes - I tried to do final check today by running "dry-run" before merging this PR and then found the change is needed.

klaskosk commented:
@mpmaruthu Could you please rebase this so it won't include vendoring changes? If you need the siteconfig conditions let me know so I can do another eco-goinfra bump

mpmaruthu commented:
@klaskosk:  As we discussed, this PR is ready to merge as I pulled changes from 'main' branch and dry-run checks passed. I will create a separate PR with minor changes later today to make test automation code more robust.

Thanks,
Peri M

klaskosk reviewed (approved):
/lgtm
We can merge after release-4.17 code freeze
